test/httpServer.cpp

```diff
@@ -15,93 +15,133 @@ uint16_t HttpServer::GetPort() {
         OnRequestOptions(conn, msg);
     } else {
-        mg_http_send_error(conn, 404, "Not Found");
+        std::string errorMessage{"Not Found"};
+        SendError(conn, 404, errorMessage);
     }
 }
 
-void HttpServer::OnRequestOptions(mg_connection* conn, http_message* /*msg*/) {
+void HttpServer::OnRequestOptions(mg_connection* conn, mg_http_message* /*msg*/) {
     std::string headers =
             "Content-Type: text/plain\r\n"
             "Access-Control-Allow-Origin: *\r\n"
             "Access-Control-Allow-Credentials: true\r\n"
             "Access-Control-Allow-Methods: GET, POST, PUT, DELETE, PATCH, OPTIONS\r\n"
-            "Access-Control-Max-Age: 3600";
+            "Access-Control-Max-Age: 3600\r\n";
 
-    mg_send_head(conn, 200, 0, headers.c_str());
     std::string response;
-    mg_send(conn, response.c_str(), response.length());
+    mg_http_reply(conn, 200, headers.c_str(), response.c_str());
 }
 
-void HttpServer::OnRequestTimeout(mg_connection* conn, http_message* msg) {
+void HttpServer::OnRequestTimeout(mg_connection* conn, mg_http_message* msg) {
     std::this_thread::sleep_for(std::chrono::milliseconds(100));
     OnRequestHello(conn, msg);
 }
 
-void HttpServer::OnRequestLowSpeedTimeout(mg_connection* conn, http_message* /*msg*/) {
-    std::string response{"Hello world!"};
-    std::string headers = "Content-Type: text/html";
-    mg_send_head(conn, 200, response.length() * 20, headers.c_str());
-    for (size_t i = 0; i < 20; i++) {
-        mg_send(conn, response.c_str(), response.length());
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    }
+void HttpServer::OnRequestLongTimeout(mg_connection* conn, mg_http_message* msg) {
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+    OnRequestHello(conn, msg);
 }
 
-void HttpServer::OnRequestLowSpeed(mg_connection* conn, http_message* /*msg*/) {
+// Send the header, then send "Hello world!" every 100ms
+// For this, we use a mongoose timer
+void HttpServer::OnRequestLowSpeedTimeout(mg_connection* conn, mg_http_message* /* msg */, TimerArg* timer_arg) {
     std::string response{"Hello world!"};
-    std::string headers = "Content-Type: text/html";
-    mg_send_head(conn, 200, response.length(), headers.c_str());
-    std::this_thread::sleep_for(std::chrono::seconds(2));
-    mg_send(conn, response.c_str(), response.length());
+    mg_printf(conn, "HTTP/1.1 200 OK\r\nContent-Type: text/html\r\nContent-Length: %d\r\n\r\n", response.length() * 20);
+    timer_arg->connection_id = conn->id;
+    mg_timer_init(
+            &timer_arg->mgr->timers, &timer_arg->timer, 100, MG_TIMER_REPEAT,
+            // The following lambda function gets executed each time the timer is called.
+            // It sends "Hello world!" to the client each 100ms at most 20 times.
+            [](void* arg) {
+                auto* timer_arg = static_cast<TimerArg*>(arg);
```

Comment:
Please try to avoid `auto` when ever possible.

```suggestion
                TimerArg* timer_arg = static_cast<TimerArg*>(arg);
```